File: src/dataset/typescript/resistor-color-duo_exercism-wackerow/gemini/GeminiFlash_resistor-color-duo_v4.ts
export class ResistorColor {
  private colors: string[];

  private static readonly colorConversion: { [i: string]: number } = {
    "black": 0,
    "brown": 1,
    "red": 2,
    "orange": 3,
    "yellow": 4,
    "green": 5,
    "blue": 6,
    "violet": 7,
    "grey": 8,
    "white": 9
  };

  constructor(colors: string[]) {
    if (colors.length < 2) {
      throw "At least two colors need to be present";
    }
    this.colors = colors;
  }

  value = (): number => {
    const color1Value = ResistorColor.colorConversion[this.colors[0]];
    const color2Value = ResistorColor.colorConversion[this.colors[1]];
    return color1Value * 10 + color2Value;
  };
}

File: src/dataset/Java/resistor-color-duo_exercism-johannesrave-education/gemini/GeminiFlash_ResistorColorDuo_v4.java
import java.util.List;

class ResistorColorDuo {
    private static final List<String> colorCode = List.of(
            "black", "brown", "red", "orange", "yellow",
            "green", "blue", "violet", "grey", "white");

    int value(String[] colors) {
        return colorCode.indexOf(colors[0]) * 10 + colorCode.indexOf(colors[1]);
    }
}